src/blob/persistence/LokiBlobMetadataStore.ts

```diff
@@ -3562,4 +4042,164 @@ export default class LokiBlobMetadataStore
+
+    if (versionIdProvided) {
+      // If versionId is provided, simply find and return that specific version
+      blobDocFindChain = blobDocFindChain.find({ versionId: versionId });
+      return blobDocFindChain.data()[0];
+    } else if (snapshotProvided) {
+      // If snapshot is provided, find that specific snapshot
+      blobDocFindChain = blobDocFindChain.find({ snapshot: snapshot });
+      return blobDocFindChain.data()[0];
+    } else if (this.isBlobVersioningEnabled()) {
+      let blobDoc = blobDocFindChain.find({ versionId: "" }).data()[0];
+
+      if (blobDoc) {
+        // This will only happen when versioning was previously disabled and is now
+        // enabled.
+        // TODO: Check Azure Prod behaviour
+        return blobDoc;
+      }
+
+      blobDocFindChain = coll.chain().find(initQuery);
+      // If versioning is enabled and no versionId/snapshot provided, return the current version
+      blobDoc = blobDocFindChain.find({ isCurrentVersion: true }).data()[0];
+
+      return blobDoc;
+    } else {
+      // If versioning is disabled and no snapshot provided
+      // First try to find blob with versionId === ""
+      const emptyVersionBlob = blobDocFindChain
+        .find({ versionId: "", snapshot: "" })
+        .data()[0];
+      if (emptyVersionBlob) {
+        return emptyVersionBlob;
+      }
+
+      blobDocFindChain = coll.chain().find(initQuery);
+      // If not found, return the current version
+      blobDocFindChain = blobDocFindChain
+        .find({
+          snapshot: ""
+        })
+        .find({ isCurrentVersion: true });
+      return blobDocFindChain.data()[0];
+    }
+  }
+
+  /**
+   * Revive well-known date fields on a single blob document.
+   * For each targeted field:
+   *  - If it's already a Date, leave it.
+   *  - If it's a non-empty string, attempt to parse as Date. If valid, replace with Date.
+   *  - Ignore null/undefined or whitespace-only strings.
+   */
+  private reviveKnownDateFields(context: Context, blob: BlobModel): BlobModel {
+    if (!blob || typeof blob !== "object") {
+      return blob;
+    }
+
+    const propDateCannotBeUndefined: Array<
+      keyof Models.BlobPropertiesInternal
+    > = ["lastModified"];
+
+    const propDateKeys: Array<keyof Models.BlobPropertiesInternal> = [
+      "creationTime",
+      "copyCompletionTime",
+      "deletedTime",
+      "accessTierChangeTime",
+      "lastAccessedOn",
+      "immutabilityPolicyExpiresOn",
+      "expiresOn"
+    ];
+
+    const topLevelDateKeys: Array<keyof IBlobAdditionalProperties> = [
+      "leaseExpireTime",
+      "leaseBreakTime"
+    ];
+
+    // Helper
+
+    if (blob.properties) {
+      for (const k of propDateCannotBeUndefined) {
+        const parsedValue = parseDateFromAssumedString(blob.properties[k]);
+        if (parsedValue) {
+          (blob.properties[k] as Date) = parsedValue;
+        } else {
+          throw StorageErrorFactory.getInvalidOperation(
```

Comment:
Could we not throw error here, but just write a debug log?
Then there will be no regression concern.